iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -115,147 +119,90 @@ storable_function<S, signature<ReturnType, Args...>>::operator=(storable_functio
 {
     storable_function tmp = std::move(f);
     f = std::move(*this);
     *this = std::move(tmp);
 }
 
-template <typename S, typename T>
-void swap(storable_function<S, T>& f, storable_function<S, T>& g) noexcept
+template <uint64_t Capacity, typename T>
+inline void swap(storable_function<Capacity, T>& f, storable_function<Capacity, T>& g) noexcept
 {
     f.swap(g);
 }
 
-template <typename S, typename ReturnType, typename... Args>
-bool storable_function<S, signature<ReturnType, Args...>>::empty() const noexcept
-{
-    return m_invoker == nullptr;
-}
-
-template <typename S, typename ReturnType, typename... Args>
+template <uint64_t Capacity, typename ReturnType, typename... Args>
 template <typename Functor, typename>
-void storable_function<S, signature<ReturnType, Args...>>::storeFunctor(const Functor& functor) noexcept
+inline void storable_function<Capacity, signature<ReturnType, Args...>>::storeFunctor(const Functor& functor) noexcept
 {
     using StoredType = typename std::remove_reference<Functor>::type;
     auto ptr = m_storage.template allocate<StoredType>();
-
-    if (ptr)
-    {
-        // functor will fit, copy it
-        /// @NOLINTJUSTIFICATION ownership is encapsulated in storable_function and will be released in destroy
-        /// @NOLINTNEXTLINE(cppcoreguidelines-owning-memory)
-        ptr = new (ptr) StoredType(functor);
-
-        // erase the functor type and store as reference to the call in storage
-        m_callable = ptr;
-        m_invoker = invoke<StoredType>;
-        m_operations.copyFunction = copy<StoredType>;
-        m_operations.moveFunction = move<StoredType>;
-        m_operations.destroyFunction = destroy<StoredType>;
-    }
-    else
-    {
-        std::cerr << "storable_function: no memory to store functor\n";
-        // this cannot happen in the static_storage case
-    }
-
-    cxx::Ensures(!empty());
+    cxx::Expects(ptr != nullptr);
+
+    /// @NOLINTJUSTIFICATION ownership is encapsulated in storable_function and will be released in destroy
+    /// @NOLINTNEXTLINE(cppcoreguidelines-owning-memory)
+    ptr = new (ptr) StoredType(functor);
+
+    // erase the functor type and store as reference to the call in storage
+    m_callable = ptr;
+    m_invoker = invoke<StoredType>;
+    m_operations.copyFunction = copy<StoredType>;
+    m_operations.moveFunction = move<StoredType>;
+    m_operations.destroyFunction = destroy<StoredType>;
 }
 
-template <typename S, typename ReturnType, typename... Args>
+template <uint64_t Capacity, typename ReturnType, typename... Args>
 template <typename CallableType>
-void storable_function<S, signature<ReturnType, Args...>>::copy(const storable_function& src,
-                                                                storable_function& dest) noexcept
+inline void storable_function<Capacity, signature<ReturnType, Args...>>::copy(const storable_function& src,
```

Comment:
The name is not ideal since one cannot just call `copy` but in the case of copy assignment one needs to also call destroy. It's the same with `move` and I had to look it up in order to be sure that there isn't a bug.

Since this is existing code it does not need to be changed in this PR but if you like, how about either renaming this to `copy_without_destroy` or similar?